### PR TITLE
chore: Enable Nunjucks throwOnUndefined

### PR DESCRIPTION
**Pull request by btmills**
This will throw if we're missing any parameters or accidentally reference substitutions that don't exist. As you'll see in these commits, it catches a number of small accidental omissions.

### CODE REVIEW

**Comment by btmills** [comment]
I was able to deduce the missing fr string. The es and ja sites are missing the [`from_the_blog`](https://github.com/eslint/eslint.org/blob/220f337003b759c2b95151f9b1c64f53c8142e86/src/_data/sites/en.yml#L285) string. Adding those will allow all sites to pass. @AlixFachin and @luifermoron, would you be able to provide translations for the string I've linked above?

**Comment by AlixFachin** [comment]
> I was able to deduce the missing fr string. The es and ja sites are missing the [`from_the_blog`](https://github.com/eslint/eslint.org/blob/220f337003b759c2b95151f9b1c64f53c8142e86/src/_data/sites/en.yml#L285) string. Adding those will allow all sites to pass. @AlixFachin and @luifermoron, would you be able to provide translations for the string I've linked above?

Sure - will have a look today!
Should I create a brand new PR with the answer?

**Comment by luifermoron** [comment]
> I was able to deduce the missing fr string. The es and ja sites are missing the [`from_the_blog`](https://github.com/eslint/eslint.org/blob/220f337003b759c2b95151f9b1c64f53c8142e86/src/_data/sites/en.yml#L285) string. Adding those will allow all sites to pass. @AlixFachin and @luifermoron, would you be able to provide translations for the string I've linked above?

Sure, I will do it. But now I am facing a COVID situation. I will work on it in following days.

**Comment by nzakas** [comment]
Nice, thanks for doing this @btmills 

@AlixFachin yes please.

@luifermoron take care of yourself.

Worst case scenario is we just use English in the short term.

**Comment by AlixFachin** [comment]
@btmills @nzakas 
Ok I added the translation for the snippet - (最新のブログ記事, which means "recent blog articles")
The PR is located there:
<https://github.com/eslint/eslint.org/pull/259>

I hope this helps!

**Comment by btmills** [comment]
Thank you, @AlixFachin - that worked! I rebased this branch now that #259 is merged, and the ja build is fixed.

The new pt-br site only had one typoed key, and it's working after a quick fix too.

I'll leave this for a few days to give luifermoron some time and add a temporary English fallback if that's not long enough.

**Comment by nzakas** [comment]
“From the blog” only ever appears on blog pages, which never show up on the translation sites. So I think we are good with just using an English placeholder.

**Comment by btmills** [comment]
Thank you, @luifermoron - rebased and everything's green!

Marked as ready for review. This is probably easiest to review one commit at a time, and I've included explanations in most commit messages.

File: src/content/pages/sponsors.html [hunk 1]
```diff
@@ -9,6 +9,7 @@
 {%- from 'components/donation.macro.html' import donationItem %}
 {%- from 'components/hero.macro.html' import hero %}
 
+{% set title = site.sponsors_page.title %}
```
Comment (nzakas) [inline src/content/pages/sponsors.html:12]: This shouldn’t be necessary? The page title is pulled using the hook value in base.html.

Comment (btmills) [reply src/content/pages/sponsors.html:12]: Disclaimer: my knowledge of Eleventy is fairly limited.

If I remove this line, `title` is undefined.

I see `page_title` being pulled based on `hook` in `base.html`:

https://github.com/eslint/eslint.org/blob/6a26fa2c26d7f332ef2e564da6d753818bda5dca/src/_includes/layouts/base.html#L14

If I remove this line and change `hero({ title: title })` to `hero({ title: page_title })`, `page_title` is also undefined. Should `page_title` be in scope here from `base.html`?

Comment (nzakas) [reply src/content/pages/sponsors.html:12]: Ah I see. In this case, the call to hero should be passing `site.sponsors_page.title` instead of just `title`.

Comment (btmills) [reply src/content/pages/sponsors.html]: Gotcha! I updated to do that in ab9854d. I also rebased on `main` to make sure this has the latest changes.

**Review by nzakas — APPROVED** [review]
LGTM. Huge win with this.

**Pull request merged**